# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/URLQueryStringStripper.h
# Commit: c2ca2f596c09
# Full Hash: c2ca2f596c0922e7fd34494dc6ed8dd68d46b122
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-03-26 04:11:24
# Regressor Bug: 1877419
# File Overlap Count: 1
# Description:
#   Bug 1877419 - Part 1: Added nested stripping of urls to copy without site tracking. r=pbz,necko-reviewers
#   
#   Depends on D204914
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202943
# ==============================================================================

diff -r 45931e6db21e -r c2ca2f596c09 toolkit/components/antitracking/URLQueryStringStripper.h
--- a/toolkit/components/antitracking/URLQueryStringStripper.h	Mon Mar 25 18:30:38 2024 +0000
+++ b/toolkit/components/antitracking/URLQueryStringStripper.h	Mon Mar 25 18:30:38 2024 +0000
@@ -49,6 +49,12 @@
   void PopulateStripList(const nsAString& aList);
   void PopulateAllowList(const nsACString& aList);
 
+  // Recursive helper function that helps strip URIs of tracking parameters
+  // and enables the stripping of tracking paramerters that are in a URI which
+  // is nested in a query parameter
+  nsresult StripForCopyOrShareInternal(nsIURI* aURI, nsIURI** strippedURI,
+                                       int& aStripCount, bool aStripNestedURIs);
+
   nsTHashSet<nsString> mList;
   nsTHashSet<nsCString> mAllowList;
   nsCOMPtr<nsIURLQueryStrippingListService> mListService;
